server/routing.go

```diff
@@ -85,6 +85,13 @@ func (svr *Server) RegisterRoutes(r *mux.Router) {
 	)
 
 	r.HandleFunc("/health", withLogging(svr.handleHealth, svr.log)).Methods("GET")
+
+	// Only register admin endpoints if explicitly enabled in configuration
+	if svr.config.AdminEndpointsEnabled {
```

Comment:
in the vain of using dynamic lists for configuration - curious if there's any value if instead taking a list of enabled "APIs"; e.g:
```
--api-enabled proxy, memconfig, admin 
```

analogous to how geth does ([e.g](https://github.com/offchainlabs/nitro-testnode/blob/release/docker-compose.yaml#L69))

Reply:
Given that it's a breaking change to how memstore is configured, and based on the complexity of this change, I don't think we should do this right now

Reply:
Don't think its a breaking change though given that proxy is auto-enabled, and memconfig api is also auto-enabled (afair) when --memstore.enabled is passed.
So this would just be a future-compatible way to enable apis (we could eventually enable the memstore api that way as well (also I think breaking changes for testing apis are totally fine)

Reply:
> we could eventually enable the memstore api that way as well

Fair point, when considering complexity and impact, I was assuming we would change how memstore works right now. But there isn't any reason we need to do that, we can simply use this new pattern for the admin API

With that in mind, I'll go ahead and implement this suggestion for the admin endpoints

Reply:
[714f107](https://github.com/Layr-Labs/eigenda-proxy/pull/360/commits/714f10781b2d380cef2d133c401bba1319a24d24)